packages/apollo-collaboration-server/src/counters/counters.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable, Logger, NotFoundException } from '@nestjs/common'
+import { InjectModel } from '@nestjs/mongoose'
+import { Counter, CounterDocument } from 'apollo-schemas'
+import { Model } from 'mongoose'
+
+@Injectable()
+export class CountersService {
+  constructor(
+    @InjectModel(Counter.name)
+    private readonly counterModel: Model<CounterDocument>,
+  ) {}
+
+  private readonly logger = new Logger(CountersService.name)
+
+  async getNextSequenceValue(sequenceName: string): Promise<number> {
```

Comment:
This function isn't behaving. It's successfully finding the sequence value but seemingly not incrementing the value when changes are made. I've attempted all alternatives for "returnNewDocument" outlined here https://stackoverflow.com/a/47918879 (e.g. "returnOriginal", etc), all to no avail.

Reply:
Found the issue, I was calling findOneAndUpdate incorrectly (passing all parameters as a dictionary in the first slot).